streams-contrib/streams-provider-twitter/src/main/java/org/apache/streams/twitter/provider/TwitterFollowingProviderTask.java

```diff
@@ -144,6 +165,57 @@ else if( endpoint.equals("friends") )
         } while (curser != 0 && keepTrying < 10);
     }
 
+    private void collectIds(Long id) {
+        int keepTrying = 0;
+
+        long curser = -1;
+
+        do
+        {
+            try
+            {
+                twitter4j.IDs ids = null;
+                if( endpoint.equals("followers") )
+                    ids = client.friendsFollowers().getFollowersIDs(id.longValue(), curser, max_per_page);
+                else if( endpoint.equals("friends") )
+                    ids = client.friendsFollowers().getFriendsIDs(id.longValue(), curser, max_per_page);
+
+                Preconditions.checkNotNull(ids);
```

Comment:
This is a little roundabout, right? The conditions where `ids == null` is where either `!endpoint.equals("followers")` or `!endpoint.equals("friends")`. Really the Preconditions should be checking *that*, and throwing a more verbose exception in the case it fails.